### PR TITLE
CLOUDP-112237: Env Variables support

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-112237

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

This PR adds support for `MONGODB_ATLAS_`  and `MCLI_` env variables for atlasCLI

**WITHOUT ENV VARs**
```
./bin/atlas cluster ls
Error: missing credentials

To set credentials, run: atlas config
```

**Using MONGODB_ATLAS_**
```
export MONGODB_ATLAS_ORG_ID=5e429e7706822c6eac4d5970
export MONGODB_ATLAS_PROJECT_ID=5e4e593f70dfbf1010295836
export MONGODB_ATLAS_OPS_MANAGER_URL=https://cloud-dev.mongodb.com/
export MONGODB_ATLAS_SERVICE=cloud
export MONGODB_ATLAS_PRIVATE_API_KEY=privateKey
export MONGODB_ATLAS_PUBLIC_API_KEY=ejihuazr
➜  ~/workspace/mongocli git:(CLOUDP-112237) ✗ ./bin/atlas cluster ls
ID                         NAME                    MDB VER   STATE
620a21fcc60a18018675c3bc   Quickstart-1644831225   5.0.6     IDLE
```

**Using MCLI_**
```
export MCLI_ORG_ID=5e429e7706822c6eac4d5970
export MCLI_PROJECT_ID=5e4e593f70dfbf1010295836
export MCLI_OPS_MANAGER_URL=https://cloud-dev.mongodb.com/
export MCLI_SERVICE=cloud
export MCLI_PRIVATE_API_KEY=privateKey
export MCLI_PUBLIC_API_KEY=ejihuazr
➜  ~/workspace/mongocli git:(CLOUDP-112237) ./bin/atlas cluster ls
ID                         NAME                    MDB VER   STATE
620a21fcc60a18018675c3bc   Quickstart-1644831225   5.0.6     IDLE
```

### CODE REVIEW

File: internal/config/profile.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ import (
 //go:generate mockgen -destination=../mocks/mock_profile.go -package=mocks github.com/mongodb/mongocli/internal/config SetSaver
 
 const (
-	EnvPrefix                    = "mcli"          // EnvPrefix prefix for ENV variables
+	MCLIEnvPrefix                = "MCLI"          // MCLIEnvPrefix prefix for MongoCLI ENV variables
```
Comment (blva) [inline internal/config/profile.go:40]: [nit] `MongoCLIEnvPrefix` to match Atlas pattern?

**Review by blva — APPROVED** [review]
LGTM, only one nit but fine to ignore

File: internal/config/profile.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/config/profile.go]: Updated

**Review by blva — APPROVED** [review]
LGTM

**Review by johansteyn — APPROVED** [review]
LGTM

**Pull request merged**